docs/data/data-grid/filtering-recipes/FilteringLocalStorage.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import {
+  DataGrid,
+  DataGridProps,
+  GridFilterModel,
+  GridToolbar,
+} from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const VISIBLE_FIELDS = ['name', 'rating', 'country', 'dateCreated', 'isAdmin'];
+
+const createFilterModelStore = () => {
+  let listeners: Array<() => void> = [];
+  const lsKey = 'gridFilterModel';
+  const emptyModel = 'null';
+
+  return {
+    subscribe: (callback: () => void) => {
+      listeners.push(callback);
+      return () => {
+        listeners = listeners.filter((listener) => listener !== callback);
+      };
+    },
+    getSnapshot: () => {
+      try {
+        return localStorage.getItem(lsKey) || emptyModel;
+      } catch (error) {
+        return emptyModel;
+      }
+    },
+
+    getServerSnapshot: () => {
+      return emptyModel;
+    },
+
+    update: (filterModel: GridFilterModel) => {
+      localStorage.setItem(lsKey, JSON.stringify(filterModel));
+      listeners.forEach((listener) => listener());
+    },
+  };
+};
+
+const usePersistedFilterModel = () => {
+  const [filterModelStore] = React.useState(createFilterModelStore);
+
+  const filterModelString = React.useSyncExternalStore(
+    filterModelStore.subscribe,
+    filterModelStore.getSnapshot,
+    filterModelStore.getServerSnapshot,
+  );
+
+  const filterModel = React.useMemo(() => {
+    try {
+      return (JSON.parse(filterModelString) as GridFilterModel) || undefined;
+    } catch (error) {
+      return undefined;
+    }
+  }, [filterModelString]);
+
+  return React.useMemo(
+    () => ({
+      filterModel,
+      setFilterModel: filterModelStore.update,
+    }),
+    [filterModel, filterModelStore.update],
+  );
+};
+
+export default function FilteringLocalStorage() {
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    visibleFields: VISIBLE_FIELDS,
+    rowLength: 100,
+  });
+
+  const { filterModel, setFilterModel } = usePersistedFilterModel();
```

Comment:
Nitpick: How about keeping its signature similar to `React.useState`?
```suggestion
  const [filterModel, setFilterModel] = usePersistedFilterModel();
```